bpf/lib/nat.h

```diff
@@ -552,6 +552,41 @@ snat_v4_process(struct __ctx_buff *ctx, enum nat_dir dir,
 			tuple.sport = 0;
 			icmp_echoreply = true;
 			break;
+		case ICMP_DEST_UNREACH:
+			switch (icmphdr.code) {
+			case ICMP_FRAG_NEEDED:
+				if (1) {
+				        struct iphdr iphdr;
+					__be16 identifier = 0;
+					__u8 type;
+
+					// RFC 5508 indicates that original
+					// IPV4+ICMP headers are added in
+					// payload.
```

Comment:
I'd like to check that my understanding is correct.

We try to deal with the following case (#12968)
```
Pod A (MTU 1500) -> Node A (MTU 1500) -> Switch X (MTU 1420) -> Node B (MTU 1500)
```
Specifically, what to do when we get a `ICMP_FRAG_NEEDED` error from, e.g., Switch X, above.


Quoting RFC5508:
```
ICMP Error Messages - ICMP Error messages provide signaling for IP.
     All ICMP Error messages are characterized by the fact that they
     embed the original datagram that triggered the ICMP Error message.
     The original datagram embedded within the ICMP Error payload is
     also referred to as the "Embedded packet" throughout the document.
     Unlike ICMP Query messages, ICMP Error messages do not have a Query
     Id in the ICMP header.
```
Wouldn't be possible for the embedded packet to be any other packet? (e.g., TCP or UDP)? 
We seem to be assuming embedded packet will always be an ICMP_ECHO. Is there a reason for this (complexity, presumably)?

Reply:
Yes you are totally right I was making wrong assumption saying that the embedded packet will always be an ICMP_ECHO (I guess for IPPROTO ICMP, it can't be anything else, right?)

I'm currently fixing that part, I should provide new PR shortly.